rasa_nlu/components.py

```diff
@@ -128,11 +127,13 @@ class Component(object):
     features that are used by another component down
     the pipeline to do intent classification."""
 
-    # Name of the component to be used when integrating it in a
+    # Component class name is used when integrating it in a
     # pipeline. E.g. ``[ComponentA, ComponentB]``
     # will be a proper pipeline definition where ``ComponentA``
     # is the name of the first component of the pipeline.
-    name = ""
+    @classmethod
+    def name(cls):
```

Comment:
I think it makes sense to do some additional effort to make this a property again to get backwards compatibility
```python
class ComponentMeta(type):
  @property
  def name(cls):
    return cls.__name__

class Component(metaclass=ComponentMeta):
  #...
```
ref https://stackoverflow.com/a/1800999

Reply:
I remember, I looked at this stack overflow discussion 😀 didn't want to bother, but I'll do that